interfaces/cython/cantera/ctml_writer.py

```diff
@@ -2163,7 +2200,148 @@ def build(self, p):
         k = ph.addChild("kinetics")
         k['model'] = 'none'
 
+class IdealSolidSolution(phase):
+    """An IdealSolidSolution phase."""
+    def __init__(self,
+                 name = '',
+                 elements = '',
+                 species = '',
+                 note = '',
+                 density = None,
+                 transport = 'None',
+                 initial_state = None,
+                 standard_concentration = None,
+                 options = []):
+
+        phase.__init__(self, name, 3, elements, species, note, 'None',
+                       initial_state, options)
+        self._pure = 0
+        self._stdconc = standard_concentration
+        if self._stdconc is None:
+            raise CTI_Error('In phase ' + name + ': standard_concentration must be specified.')
+        self._tr = transport
+
+    def conc_dim(self):
+        return (1,-3)
 
+    def build(self, p):
+        ph = phase.build(self, p)
+        e = ph.child("thermo")
+        e['model'] = 'IdealSolidSolution'
+        if self._tr:
+            t = ph.addChild('transport')
+            t['model'] = self._tr
+        k = ph.addChild("kinetics")
+        k['model'] = 'none'
+        sc = ph.addChild('standardConc')
+        sc['model'] = self._stdconc
+                
+class BinarySolutionTabulatedThermo(phase):
+    """A BinarySolutionTabulatedThermo phase."""
+    def __init__(self,
+                 name = '',
+                 elements = '',
+                 species = '',
+                 note = '',
+                 transport = 'None',
+                 initial_state = None,
+                 standard_concentration = None,
+                 tabulated_species = '',
+                 tabulated_thermo = None,
+                 options = []):
+        phase.__init__(self, name, 3, elements, species, note, 'None',
+                       initial_state, options)
+        self._pure = 0
+        self._tabSpecies = tabulated_species
+        self._tabThermo = tabulated_thermo
+        self._stdconc = standard_concentration
+        self._tr = transport
+        if self._stdconc is None:
+            raise CTI_Error('In phase ' + name + ': standard_concentration must be specified.')
+        if tabulated_species is None:
+            raise CTI_Error('In phase ' + name + ': tabulated_species must be specified.')
+        if tabulated_thermo is None:
+            raise CTI_Error('In phase ' + name + ': A tabulated thermo data must be specified.')
+        
+    def conc_dim(self):
+        return (1,-3)
+    def build(self, p):
+        ph = phase.build(self, p)
+        e = ph.child("thermo")
+        e['model'] = 'BinarySolutionTabulatedThermo'
+        if self._tabSpecies:
+            e1 = e.addChild('tabulatedSpecies')
+            e1['name'] = self._tabSpecies
+        if self._tabThermo:
+            t = e.addChild("tabulatedThermo")
+            if isinstance(self._tabThermo, thermo):
+                self._tabThermo.build(t)
```

Comment:
Does it matter if `_tabThermo` is a `thermo` instance or not? As long as it has an appropriate `build` method, that should be fine. And if it doesn't, the alternate case of silently skipping it doesn't seem better than just letting it fail with whatever exception otherwise would have occurred.